### ISSUE TITLE
Invalid "unknown prop" warnings for SSR

### ISSUE THREAD

**Issue by bvaughn** [facebook/react#10265]
Run the [`fixtures/ssr` test harness](https://github.com/facebook/react/tree/master/fixtures/ssr). In the terminal, the following warning is logged:

```
Warning: Unknown prop `onClick` on <a> tag. Remove this prop from the element. For details, see https://fb.me/react-unknown-prop
  in a (at Page.js:12)
  in p (at Page.js:18)
  in div (at Page.js:17)
  in Page (at App.js:12)
  in div (at App.js:10)
  in body (at Chrome.js:17)
  in html (at Chrome.js:9)
  in Chrome (at App.js:9)
  in App (at render.js:18)
```

SSR behavior should match client-side behavior and not warn for a valid attribute.

**Comment by gzhappysky** [comment]
Because ['ReactDOMClientInjection'](https://github.com/facebook/react/blob/master/src/renderers/dom/shared/ReactDOMClientInjection.js#L40) does not required in ['ReactDOMServerEntry'](https://github.com/facebook/react/blob/master/src/renderers/dom/ReactDOMServerEntry.js#L17). so that Event plugin does not loaded, [This line](https://github.com/facebook/react/blob/master/src/renderers/dom/shared/hooks/ReactDOMUnknownPropertyHook.js#L67) does not validate passed.

 I think we might require ReactDOMClientInjection in [ReactDOMServerEntry](https://github.com/facebook/react/blob/master/src/renderers/dom/ReactDOMServerEntry.js) and load [event plugin](https://github.com/facebook/react/blob/master/src/renderers/dom/shared/ReactDOMClientInjection.js#L40) to solve this problem.

 add the follow line to [ReactDOMServerEntry](https://github.com/facebook/react/blob/master/src/renderers/dom/ReactDOMServerEntry.js) .

`require('ReactDOMClientInjection');`

**Comment by gaearon** [comment]
Oh I see. We don't want the client injection to keep server renderer small. But I realize now it needs the event system in DEV. This is interesting.

**Comment by gzhappysky** [comment]
👍 , Are we only injection event system plugin? 

 

> EventPluginHub.injection.injectEventPluginsByName({
  SimpleEventPlugin: SimpleEventPlugin,
  EnterLeaveEventPlugin: EnterLeaveEventPlugin,
  ChangeEventPlugin: ChangeEventPlugin,
  SelectEventPlugin: SelectEventPlugin,
  BeforeInputEventPlugin: BeforeInputEventPlugin,
});

**Comment by gaearon** [comment]
@gzhappysky Thanks for diagnosing where this comes from. I think I have a fix in #10272.
